### PR TITLE
Add UNO R4 WiFi basic support

### PR DESCRIPTION
**Pull request by pennam**
To connect to ArduinoIoTCloud WiFi firmware need to be updated. The minimum required version is `0.2.0`.

Pre release of the firmware is available here: https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/releases
Updater tool for [linux](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/releases/download/0.2.0/unor4wifi-update-linux.zip), [win](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/releases/download/0.2.0/unor4wifi-update-windows.zip), [macos](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/releases/download/0.2.0/unor4wifi-update-macos.zip).

Instructions for flashing the new firmware available [here](https://github.com/bcmi-labs/uno-r4-wifi-usb-bridge/blob/b1ddb36decd16f348e4e5796b2040c4b94b7e821/unor4wifi-updater/README.md)

### CODE REVIEW

**Comment by per1234** [comment]
About the failure of the "Unit Tests" workflow run:
https://github.com/arduino-libraries/ArduinoIoTCloud/actions/runs/5398721798/jobs/9804897397?pr=366#step:4:49

We have been seeing a recent increase in these spurious failures caused by intermittent errors while uploading the coverage data to Codecov.

The Tooling Team has had success by adding the upload token as suggested in the error message:
https://github.com/arduino/arduino-cli/pull/2129

There is a detailed explanation of our chosen approach in the commit message here:
https://github.com/arduino/report-size-deltas/commit/417507116d94cde5afd3ef72d9f77dc115467407

If you think that approach would be appropriate for this project, I will be happy to submit a pull request to make the equivalent change in the "Unit Tests" workflow.

**Comment by pennam** [comment]
> If you think that approach would be appropriate for this project, I will be happy to submit a pull request to make the equivalent change in the "Unit Tests" workflow.

Thanks @per1234 that would be perfect!

**Comment by pennam** [comment]
rebased to fix conflicts

File: src/ArduinoIoTCloudTCP.cpp [hunk 1]
```diff
@@ -327,6 +327,12 @@ ArduinoIoTCloudTCP::State ArduinoIoTCloudTCP::handle_ConnectMqttBroker()
   reconnection_retry_delay = min(reconnection_retry_delay, static_cast<unsigned long>(AIOT_CONFIG_MAX_RECONNECTION_RETRY_DELAY_ms));
   _next_connection_attempt_tick = millis() + reconnection_retry_delay;
 
+#if defined(ARDUINO_UNOWIFIR4)
+    if (String(WiFi.firmwareVersion()) < String("0.2.0")) {
+      DEBUG_ERROR("ArduinoIoTCloudTCP::%s In order to connect to Arduino IoT Cloud, WiFI firmware needs to be >= 0.2.0, current %s", __FUNCTION__, WiFi.firmwareVersion());
```
Comment (csarnataro) [inline src/ArduinoIoTCloudTCP.cpp:332]: It should read `WiFi firmware` instead of `WiFI firmware` (with lowercase i)

**Pull request merged**